programs/zstdcli.c

```diff
@@ -409,8 +449,20 @@ static int readSizeTFromCharChecked(const char** stringPtr, size_t* value)
         if (**stringPtr=='i') (*stringPtr)++;
         if (**stringPtr=='B') (*stringPtr)++;
     }
-    *value = result;
-    return 0;
+
+    //Only well formed entries will be 0 or , (specifically for zstd compound)
+    if (**stringPtr == 0 || **stringPtr == ',') {
+
+        *value = result;
+        return 0;
+    
+    }else{
+
+        *value = 0;
```

Comment:
This is an error, in such a case we don't want to change `*value`.

Reply:
Understood - so error out similar to overflow?

Reply:
Yes, but this is the internal function that doesn't error out. Even so, lets keep behavior consistent in the case of errors and not assign anything to `*value` on error.